conans/test/functional/conan_api/curdir_kept_test.py

```diff
@@ -20,7 +20,11 @@ class Pkg(ConanFile):
             # Needed to not write in the real computer cache
             with tools.environment_append({"CONAN_USER_HOME": tmp_folder}):
                 api, _, _ = ConanAPIV1.factory()
-                api.create(".", name="lib", version="1.0", user="user", channel="channel")
+                api.create(".", name="lib", version="1.0", user="user", channel="channel",
+                           profile_build=ProfileInfo(None, None, None, None),
```

Comment:
Only one doubt about this: `ProfileInfo` is just a `namedtuple` to group together `settings`, `options`, `env` and `profile`... I think a tuple is almost a primitive type, and I think it is useful to have all those four items together. If not, we are going to have 12 arguments (it was proposed to preserve the distinction between explicit `host` and old behavior) instead of 3.

But we have plenty of time to think about this, the command line interface cannot be the first PR related to this feature.